sdk/python/tests/integration/e2e/test_universal_e2e.py

```diff
@@ -45,36 +53,62 @@ def check_offline_and_online_features(
             assert abs(response_dict[f"{fv.name}__value"][0] - expected_value) < 1e-6
         else:
             assert response_dict[f"{fv.name}__value"][0] is None
+        if expected_entityless_value:
```

Comment:
Is there any way in which we can avoid more and more conditional logic in our tests? It makes understanding the tests very hard.

Reply:
Not sure what the best way to refactor into different tests would be, but I definitely agree tests keep having more conditional logic. Happy to hear @achals' thoughts though, I think the refactoring of the other 2 `test_universal_historical_retrieval.py` and `test_universal_online.py` are both great (and maybe it's the case that we don't need to make any changes to this file if the coverage there is good enough for the entityless case)